### ISSUE TITLE
Error "BLOB is not found" while replication converts INSERT into UPDATE for a conflicting record

### ISSUE THREAD

**Issue by lwennekers** [FirebirdSQL/firebird#7070]
Hello all,

I'm trying  your replication engine on Firebird 4.0.0.2496 and I've got some strange issues (perhaps wanted by your design). The following
configurations are used

MASTER configuration
database = C:\Data\Databases\Master.FDB
{
  journal_directory = C:\Data\Replication\Journal
  journal_archive_directory = C:\Data\Replication\JournalArchive
  journal_archive_timeout = 10
  disable_on_error = false
}

REPLICA configuration
database = C:\Data\Databases\REPLICA.FDB
{
  journal_source_directory = \\192.168.1.1\JournalArchive
  apply_idle_timeout = 10
  apply_error_timeout = 10
}

The replica database is configured as read_write. 

The replication is correctly function at start by I wanted to test/know what will happen is the REPLICA.FDB already has an existing record while the master is going to insert a record in the table with the same primary key.
I get a correct log-file at that moment
WARNING: Record being inserted into table History already exists, updating instead

The next line is
ERROR: Blob 227.139558 is not found for table History

At that moment the replication is going to retry to insert the record constantly and the following line is repeating every 10 seconds
WARNING: Record being inserted into table History already exists, updating instead

I was expecting the record to be updated as it is mentioned in the log-file and the next sequence to be processed. At that moment the replication is stopped and the journal files are increasing (MASTER.FDB.journal-0000000xx) while nothing is replicated anymore.

I'm now wondering what I'm missing so the replication is not stopping the process

Thanks already in advance for your support
Leonard Wennekers

**Comment by aafemt** [comment]
May be already fixed in snapshots by https://github.com/FirebirdSQL/firebird/commit/44f48955c250193096c244bee9e5cd7ddf9a099b

**Comment by lwennekers** [comment]
Thank you very much for the fast response. I presume that I can test the latest snapshot from last night.
Other questions on that: Are snapshots mature enough to use? Are you planning a new release soon?
Thanks again

**Comment by aafemt** [comment]
You can see its "maturity" on https://firebirdtest.com

**Comment by lwennekers** [comment]
I've installed you latest snapshot and tried my situation again. The problem remained and the log-file grows every 10 seconds with the message
WARNING: Record being inserted into table History already exists, updating instead

No updating is performed. Although I'm not a C++ programmer I looked into you code at the ReplServer.cpp.  When I set
verbose_logging = true I also see 
WARNING: Suspending for 10 seconds

So the exception was raised which I can understand. I then also expect that the journal files will be deleted. Is that done by the statements:
while (queue.hasData())
     delete queue.pop();

Because I can see that the journal files are remaining and at that point the sequence will start all over I presume. Hopefully this will help you and thanks again
